ote/src/cljs/ote/app/controller/transit_visualization.cljs

```diff
@@ -220,31 +220,52 @@
      ;; Add all received routes to shown map
      [:compare :show-route-lines] merge (zipmap route-line-names (repeat true)))))
 
+(defn compare-stop-differences [transit-visualization date1-trips date2-trips]
+  (if-let [first-common-stop (transit-changes/first-common-stop (concat date1-trips date2-trips))]
+          (let [first-common-stop
+                #(assoc %
+                   :first-common-stop first-common-stop
+                   :first-common-stop-time (transit-changes/time-for-stop % first-common-stop))
+                date1-trips (mapv first-common-stop date1-trips)
+                date2-trips (mapv first-common-stop date2-trips)
+                combined-trips (transit-changes/merge-by-closest-time
+                                 :first-common-stop-time
+                                 date1-trips date2-trips)]
+
+            (assoc-in transit-visualization [:compare :combined-trips]
+                      (mapv (fn [[l r]]
+                              [l r (transit-changes/trip-stop-differences l r)])
+                            combined-trips)))
+
+          ;; Can't find common stop
+          (assoc-in transit-visualization [:compare :combined-trips] nil)))
+
+
 (defn combine-trips [transit-visualization]
   (let [date1-trips (get-in transit-visualization [:compare :date1-trips])
         date2-trips (get-in transit-visualization [:compare :date2-trips])]
-
-  (if (and date1-trips date2-trips)
-    (if-let [first-common-stop (transit-changes/first-common-stop (concat date1-trips date2-trips))]
-      (let [first-common-stop
-            #(assoc %
-                    :first-common-stop first-common-stop
-                    :first-common-stop-time (transit-changes/time-for-stop % first-common-stop))
-            date1-trips (mapv first-common-stop date1-trips)
-            date2-trips (mapv first-common-stop date2-trips)
-            combined-trips (transit-changes/merge-by-closest-time
-                            :first-common-stop-time
-                            date1-trips date2-trips)]
-        (assoc-in transit-visualization [:compare :combined-trips]
-               (mapv (fn [[l r]]
-                       [l r (transit-changes/trip-stop-differences l r)])
-                     combined-trips)))
-
-      ;; Can't find common stop
-      (assoc-in transit-visualization [:compare :combined-trips] nil))
-
-    ;; Both dates not fetched, don't try to calculate
-    (assoc-in transit-visualization [:compare :combined-trips] nil))))
+    (cond
```

Comment:
Both 1. and 2. condition are almost identical and test if loading is still ongoing. If so, skip anything in this condition. This could be a good case for nested ifs, actually, since then you wouldn't have these massive condition clauses interleaved with short bits of code. Something like
(if still-loading transit-visualization (if (not-empty? date2-trips.....

Reply:
It is so much cleaner and shorter this way. Thanx, changed.